File: backend/app/schemas/ai.py
from datetime import datetime
from typing import Any, List, Optional, Dict
from pydantic import BaseModel, ConfigDict

class AIRequest(BaseModel):
    prompt: Optional[str] = None

class SummaryOut(BaseModel):
    id: int
    document_id: int
    content: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class MindmapOut(BaseModel):
    id: int
    document_id: int
    content: Dict[str, Any]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class FlashcardGenerateRequest(BaseModel):
    count: Optional[int] = 5

class FlashcardOut(BaseModel):
    id: int
    document_id: int
    front: str
    back: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class AIAskRequest(BaseModel):
    question: str

class AIAskResponse(BaseModel):
    answer: str

File: frontend/src/services/ai.ts
import api from './api';

export interface SummaryOut {
  id: number;
  document_id: number;
  content: string;
  created_at: string;
}

export interface MindmapOut {
  id: number;
  document_id: number;
  content: any;
  created_at: string;
}

export interface FlashcardOut {
  id: number;
  document_id: number;
  user_id: number;
  front: string;
  back: string;
  difficulty: string;
  created_at: string;
}

export const aiService = {
  async generateSummary(documentId: number): Promise<SummaryOut> {
    const response = await api.post(`/api/v1/ai/summary/${documentId}`);
    return response.data;
  },

  async generateMindmap(documentId: number): Promise<MindmapOut> {
    const response = await api.post(`/api/v1/ai/mindmap/${documentId}`);
    return response.data;
  },

  async generateFlashcards(documentId: number, count: number = 5): Promise<FlashcardOut[]> {
    const response = await api.post(`/api/v1/ai/flashcards/generate/${documentId}`, { count });
    return response.data;
  }
};
